src/shims/unix/fs.rs

```diff
@@ -1050,7 +1051,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             this.reject_in_isolation("`readdir`", reject_with)?;
             let eacc = this.eval_libc("EBADF");
             this.set_last_error(eacc)?;
-            return Ok(Scalar::null_ptr(this));
+            return this.write_scalar(eacc, dest);
```

Comment:
This changes behavior, doesn't it? We should be returning a null pointer but you made it return `eacc` now.

Reply:
I should fix this one in a separate PR with a test that shows the behaviour change. The previous behaviour is buggy